src/components/NewCommentForm.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { ErrorMessages } from '../types/ErrorMessages';
+import { Comment } from '../types/Comment';
+import { addComment } from '../api/coments';
+
+type Props = {
+  postId: number | undefined,
+  createNewComment: (comment: Comment) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  createNewComment,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isValidateError, setIsValidateError] = useState(false);
+
+  const [formValue, setFormValue] = useState<Partial<Comment>>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const { name, email, body } = formValue;
+
+  const handleChange = (e:React
+    .ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setFormValue((prevState) => {
+      return {
+        ...prevState,
+        [e.target.name]: e.target.value,
+      };
+    });
+  };
+
+  const handleClear = () => {
+    setFormValue(() => {
+      return {
+        name: '',
+        email: '',
+        body: '',
+      };
+    });
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (name?.length === 0 || email?.length === 0 || body?.length === 0) {
```

Comment:
Use logical operators
```suggestion
    if (!name || !email || !body) {
```